get buy links by scraping the google books website

Shivam60 opened the pull request.

Shivam60 committed:
get buy links by scraping the google books website

ahegazy commented on src/store/links.js:
```diff
@@ -0,0 +1,31 @@
+const axios = require('axios');
+const cheerio = require('cheerio');
+
```
the link.js shouldn't be in store directory .. it should with the rest of assets we are using .. like ' 	xml2json.min.js' at [src](https://github.com/chingu-voyage5/Geckos-Team-34/tree/master/src) or we can create a separate directory called 'libs' for both of them and any other thing we use .. 
we should look into that .. 
anyway please move this file to 'src'

ahegazy replied:
@Shivam60  put the file in  [dist](https://github.com/chingu-voyage5/Geckos-Team-34/tree/master/src/dist)

Shivam60 replied:
I have moved it into src/dist. Would that be fine?

ahegazy replied:
Yes .. Change  the variables names, add goodreads support and push it to your branch ..

ahegazy commented on src/store/actions.js:
```diff
@@ -2,6 +2,8 @@ import Vue from 'vue';
 import axios from 'axios'
 import dotenv from 'dotenv'
 
+var obj = require('./links.js');
```
can you name the file and the object a descriptive name ...

ahegazy commented on src/store/actions.js:
```diff
@@ -158,6 +160,11 @@ export default{
           var authors = [];
+          
+          obj.getlinks(url).then(function(res){
+						ResultsObj.links=res;
```
it would be better if it's called 'ResultsObj.buy_links' for example

Shivam60 committed:
added buylinksAPI.js

Shivam60 committed:
added buylinksAPI usage to action.js

Shivam60 commented:
I have incorporated the changes. Furthermore I have added the goodreads functionality also. So now i ll scrape the goodreads website for links and i ll get the list of urls. These urls later redirect to the respective shops such as amazon. I then return the re directed  urls.
Kindly have a look at the modifications and also check if it works as i have checked the indivisual components on my local machine only.

ahegazy commented:
I'll take a look at it and test it tomorrow.... And I'll push some  minor updates+ use the result in book.vue to your branch ....

ahegazy committed:
updates and fixes

ahegazy commented:
@Shivam60  so i made couple of updates and fixes but we have a major issue ... 
those  sites you are scrapping doesn't support that in frontend ... obviously you have been testing in the backend so you didn't face this issue ... 
I've tried to fix that using the heuroko app but i failed .. so kindly download the latest commit i made to your branch .. and run the app using 'npm run dev' and test the function of it.. .
I'll write some comments on the edits i made here..
